### ISSUE TITLE
Implement proportional + padding sampling methodology

### ISSUE THREAD

**Issue by crispy-wonton** [nestauk/asf_heat_pump_suitability#503]
For training / validation set: take the proportional + padding forward, with a defined floor of ~50 samples per primary group (to get to 1200 samples), then sample proportionally for the remaining 1250 samples.

For the test set: sample purely proportionally from the whole population across the primary and secondary constraints.
